src/core/public/application/application_service.tsx

```diff
@@ -259,6 +262,21 @@ export class ApplicationService {
       const shouldNavigate = navigatingToSameApp ? true : await this.shouldNavigate(overlays);
 
       if (shouldNavigate) {
+        const targetApp = applications$.value.get(appId);
+        if (
+          workspaces.currentWorkspaceId$.value &&
+          targetApp?.visibility === AppVisibility.homeOnly
+        ) {
+          // If user is inside a workspace and the target app is homeOnly
+          // refresh the page by doing a hard navigation
+          window.location.assign(
+            http.basePath.prepend(getAppUrl(availableMounters, appId, path), {
+              withoutClientBasePath: true,
```

Comment:
can we add a comment for `withoutClientBasePath`, i guess it means don't append workspace into path

Reply:
Sure.